queries.sql

```diff
@@ -92,4 +92,51 @@ SELECT a.* FROM animals a Join owners o ON a.owner_id = o.id WHERE
 /* Who owns the most animals? */
 SELECT count(*) as total, o.full_name as OwnerName FROM 
 	animals a JOIN owners o ON a.owner_id = o.id 
-	GROUP BY o.full_name ORDER BY total desc LIMIT 1
\ No newline at end of file
+	GROUP BY o.full_name ORDER BY total desc LIMIT 1
+
+
+  /* PROJECT DAY 4 */
+
+/*  Who was the last animal seen by William Tatcher? */
+  SELECT name FROM animals a JOIN visits v ON a.id = v.animal_id 
+	WHERE vet_id IN (SELECT id FROM vets WHERE name = 'William Tatcher') 
+	ORDER BY v.date_of_visit DESC LIMIT 1
+
+  /* How many different animals did Stephanie Mendez see? */
+  SELECT COUNT(*) FROM animals a INNER JOIN visits v ON a.id = v.animal_id 
+	WHERE vet_id IN (SELECT id FROM vets WHERE name = 'Stephanie Mendez')
+
+  /* List all vets and their specialties, including vets with no specialties. */
+  SELECT v.name, sp.name FROM vets v LEFT JOIN specializations s 
+	ON v.id = s.vet_id  LEFT JOIN species sp ON sp.id = s.species_id
+
+  /* List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020. */
+  SELECT a.name FROM animals a INNER JOIN visits v ON a.id = v.animal_id 
+	WHERE v.vet_id IN (SELECT id FROM vets WHERE name = 'Stephanie Mendez') 
+	AND v.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30'
+
+  /* What animal has the most visits to vets? */
+  SELECT a.name, COUNT(animal_id) as visits FROM animals a JOIN visits v 
+	ON a.id = v.animal_id GROUP BY a.name ORDER BY visits DESC LIMIT 1
+
+  /* Who was Maisy Smith's first visit? */
+  SELECT a.name FROM animals a INNER JOIN visits v ON a.id = v.animal_id 
+	WHERE v.vet_id IN (SELECT id FROM vets WHERE name = 'Maisy Smith') ORDER BY v.date_of_visit ASC LIMIT 1
+
+  /* Details for most recent visit: animal information, vet information, and date of visit. */
+  SELECT a.id, a.name, a.date_of_birth, a.escape_attempts, a.neutered, a.weight_kg, v.name as vet_name,
+v.age as vet_age, v.date_of_graduation as vet_graduation_date, vs.date_of_visit FROM animals a INNER
+JOIN visits vs ON a.id = vs.animal_id INNER JOIN vets v ON vs.vet_id = v.id ORDER BY vs.date_of_visit DESC LIMIT 1
+
+/* How many visits were with a vet that did not specialize in that animal's species? */
+SELECT COUNT(*)
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+JOIN species s ON a.species_id = s.id
+JOIN vets vt ON v.vet_id = vt.id
+LEFT JOIN specializations sp ON vt.id = sp.vet_id AND s.id = sp.species_id
+WHERE sp.id IS NULL
+
+/* What specialty should Maisy Smith consider getting? Look for the species she gets the most. */
+SELECT s.name FROM specializations sp INNER JOIN species s ON s.id = sp.species_id 
+	GROUP BY s.name ORDER BY COUNT(*) DESC LIMIT 1
```

Comment:
- [ ] After making all the changes in your database it would be great if you take new screenshots of these queries of the updated data and attach them in place of the older screenshots to make the next reviewer verify that all the changes are implemented correctly 👍

Reply:
okay. Thanks, @HAMMAS-SALEEM  for your effort, time and painstaking pointing out all these errors. I will correct them right away.